### PR TITLE
pre-commit: add hook to trim trailing whitespace

### PR DESCRIPTION
**Pull request by jbampton**
https://github.com/pre-commit/pre-commit-hooks#trailing-whitespace

### Description

This PR adds another hook or test to our pre-commit framework.

This hook automatically trims trailing whitespace when run locally.

Follow up PRs can target more file types if needed.

Extra unneeded whitespace is extra data.  Save the planet :)

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [X] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

`pre-commit run --all-files`

or

`pre-commit run trailing-whitespace --all-files`

### CODE REVIEW

File: test/scripts/deploy-and-run-regression.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 
 
- 
+
```
Comment (DaanHoogland) [inline test/scripts/deploy-and-run-regression.sh:21]: we could even remove some lines here
```suggestion
```

Comment (DaanHoogland) [reply test/scripts/deploy-and-run-regression.sh:21]: but maybe that is another pre-commit hook

**Review by DaanHoogland — APPROVED** [review]
looks generally good to me. Just to be clear and prevent confusion @jbampton , this is just a check and not a pre-commit corrective edit this hook is doing right?

**Comment by jbampton** [comment]
Hey @DaanHoogland  when the hook runs on GitHub with actions it is just a check that will either pass or fail.

But when pre-commit is setup locally and run this hook will also automatically trim trailing whitespace.  

So yes it is a corrective edit.  So all these whitespace clean ups were done automatically by the hook.

File: test/scripts/deploy-and-run-regression.sh (hunk 1 above)
Comment (jbampton) [reply test/scripts/deploy-and-run-regression.sh:21]: There are a large number hooks seen here: https://pre-commit.com/hooks.html

I can clean up more but I was trying to keep this pr focused on one type of issue.

**Comment by jbampton** [comment]
@blueorangutan package

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -56,7 +56,7 @@ repos:
       - id: mixed-line-ending
         exclude: \.(cs|xml)$
       - id: trailing-whitespace
-        files: \.(java|md|py|sh|vue|yaml|yml)$
+        files: \.(in|java|md|py|rb|sh|vue|yaml|yml)$
```
Comment (DaanHoogland) [inline .pre-commit-config.yaml:59]: ruby files?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by jbampton** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by vishesh92 — APPROVED** [review]
clgtm

**Pull request merged**